packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -447,6 +449,31 @@ export class ConstructNode {
     return Array.from(ret);
   }
 
+  /**
+   * Add an ordering dependency on another Construct.
+   *
+   * All constructs in the dependency's scope will be deployed before any
+   * construct in this construct's scope.
+   */
+  public addDependency(...dependencies: IConstruct[]) {
+    for (const dependency of dependencies) {
+      this.dependencies.add(dependency);
+    }
+  }
+
+  /**
+   * Return all dependencies registered on this node and its ancestors.
+   */
+  public myDependencies(): IConstruct[] {
```

Comment:
I wonder the api can be more explicit and/or rich to indicate that it returns both explicit and implicit dependencies (maybe “findDependencies(includeImpicit = true)”

Reply:
We can, but there's no use for it (at least today). Future proofing?

Reply:
I just don't like the fact that it returns implicit dependencies without that reflecting in the API. It could be `findImplicitAndExplicitDependencies()` but `findDependencies({ implicit: true })` feels a bit less ugly.

...I am not sure about the "implicit" terminology. Maybe "inherited" or "derived" or something like that...